autospec/tarball.py

```diff
@@ -43,12 +41,107 @@
 tarball_prefix = ""
 gcov_file = ""
 archives = []
-go_archives = []
 giturl = ""
 domain = ""
 prefixes = dict()
 
 
+class Source(object):
+    """Holds data and methods for source code or archives management."""
+
+    def __init__(self, url, destination, path):
+        """Set default values for source file."""
+        self.url = url
+        self.destination = destination
+        self.path = path
+        self.type = None
+        self.prefix = None
+        self.subdir = None
+
+        # Extra  compressed archives
+        if not self.destination.startswith(':'):
+            self.get_type()
+            self.get_prefix()
+            self.extract()
```

Comment:
I think I'd like if extract wasn't done as part of creating the source object since this has an effect outside the object itself and instead somebody can get the created source object (optionally inspect it even) and then call extract but I'm not sold on that being useful so I am open to keeping it this way.

Reply:
Mmm... think you are right. Maybe adding the extraction routine in inspection code would have more sense. I'll review that code and bring a new proposal.

Reply:
Alrighty, sounds good to me.

Reply:
Well, I ended up putting the extraction as part of tarball.process(). Still using Source methods, but not calling extract() as part of the object creation. If you are ok with this, I can also push the changes on test_tarball.py to fix the error and add coverage.

Reply:
Looks reasonable to me.